File: week3/이지훈/음양더하기.java
public class 음양더하기 {
    public int solution(int[] absolutes, boolean[] signs) {
        int result = 0;

        for (int i = 0; i < absolutes.length; i++) {
            if (signs[i]) result += absolutes[i];
            else result += absolutes[i] * -1;
        }

        return result;
    }

}

File: week3/한지희/음양 더하기.kt
class NegativePositivePlus {
    fun solution(absolutes: IntArray, signs: BooleanArray): Int {
        var sum = 0
        absolutes.forEachIndexed { index, number ->
            sum += if (signs[index]) number else -number
        }
        return sum
    }

    fun bestSolution(absolutes: IntArray, signs: BooleanArray) =
        absolutes.foldIndexed(0) { index, accumulator, number ->
            accumulator + if (signs[index]) number else -number
        }
}